### PR TITLE
`affeq`: Fix array OOB in `invariant`

### PR DESCRIPTION
**Pull request by michael-schwarz**
During benchmarking for #1297, it became apparent that the `invariant` function in `affeq` accesses arrays out-of-bounds due to an off-by-one error.

This fixes the issue and adds a test that has witness generation enabled to spot regressions.

### CODE REVIEW

**Review by sim642 — APPROVED** [review]
Seems right, although the entire `invariant` is very roundabout: `Matrix` → `earray` → `array` → `Enum` → `list`. A single `List.init` should do it.

File: tests/regression/63-affeq/19-witness.c [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+void main(void) {
+    int i;
+    int j;
+    int k;
+    i = 2;
+    j = k + 5;
+
+    while (i < 100) {
+        __goblint_check(3 * i - j + k == 1);
```
Comment (sim642) [inline tests/regression/63-affeq/19-witness.c:13]: This invariant doesn't appear in `witness.yml` though.

Comment (michael-schwarz) [reply tests/regression/63-affeq/19-witness.c:13]: That is another unrelated issue. The invariant is given as `i#318-1/3j#319+1/3k#320+4/3=0` which then likely does not survive conversion to a CIL expression. I suggest we open a new issue, as this is a general problem of `relationAnalysis`, which I won't have time to address any time in the next two weeks.

No longer crashing here is a step in right direction in the meanwhile.

**Review by jerhard — APPROVED** [review]

File: tests/regression/63-affeq/19-witness.c (hunk 1 above)
Comment (michael-schwarz) [reply tests/regression/63-affeq/19-witness.c:13]: I created a new issue for this.

**Pull request merged**